packages/react/src/bags/hooks/__tests__/useAddOrUpdateBagItem.test.tsx

```diff
@@ -0,0 +1,62 @@
+import { cleanup } from '@testing-library/react';
+import { getBagItems } from '@farfetch/blackout-redux/bags';
+import { mockBagItem, mockInitialState } from 'tests/__fixtures__/bags';
+import { mockStore } from '../../../../tests/helpers';
+import { Provider } from 'react-redux';
+import { renderHook } from '@testing-library/react-hooks';
+import { useAddOrUpdateBagItem } from '../..';
+import React from 'react';
+
+jest.mock('@farfetch/blackout-redux/bags', () => ({
+  ...jest.requireActual('@farfetch/blackout-redux/bags'),
+  addBagItem: jest.fn(() => ({ type: 'add' })),
+  updateBagItem: jest.fn(() => ({ type: 'update' })),
+  getBagItems: jest.fn(() => [mockBagItem]),
+}));
+
+const mockDispatch = jest.fn();
+
+jest.mock('react-redux', () => ({
+  ...jest.requireActual('react-redux'),
+  useDispatch: () => mockDispatch,
+}));
+
+const getRenderedHook = (state = mockInitialState) => {
+  const {
+    result: { current },
+  } = renderHook(() => useAddOrUpdateBagItem(mockBagItem), {
+    wrapper: props => <Provider store={mockStore(state)} {...props} />,
+  });
+
+  return current;
+};
+
+describe('useAddOrUpdateBagItem', () => {
+  beforeEach(jest.clearAllMocks);
+  afterEach(cleanup);
+
+  it('should return values correctly with initial state', () => {
+    const current = getRenderedHook();
+
+    expect(current).toStrictEqual(expect.any(Function));
+  });
+
+  describe('handleAddOrUpdateItem', () => {
```

Comment:
Do you think it's possible to add a test case for when the item has multiple merchants and it performs an update and an add operation? Since this is the main purpose of this hook, we could probably add a test ensuring exactly that 🙂

Reply:
I have added a test, can you please check it out?